### DISCUSSION TITLE
Dependent Field

### DISCUSSION THREAD

**Discussion by arikbidny** [react-hook-form/react-hook-form#3507] [Q&A] [answered]
Hey,
I have a form with : 
My first field is select with a lot of properties that comes from props, 
When I'm choosing the property from select field I want to go do API request ( Redux action ) and  render rest of selects in the form dependent on this select , 
I can't find proper and simple way to do this, 
Do you have some examples for this scenario ?

**Comment by bluebill1049** [comment, marked as answer]
check out the following API:

https://react-hook-form.com/api#watch
https://react-hook-form.com/api#useWatch

**Reply by arikbidny** [reply]
Thanks , 
Which one of them better to use for my scenario ?

**Reply by bluebill1049** [reply]
you can use watch for small form, useWatch is made for large and complex form.
